run.py

```diff
@@ -409,31 +411,80 @@ def run_diffusion_processsing(**args):
+
+                # check if length of lists in dictionary are the same
+                n = len(dwi_dict['bvalFile'])
+                assert all(len(dwi_dict[k]) for k,v in dwi_dict.items())
+
+                for dirnum in set(dwi_dict['bval']):
+                    idxs = { i for k,v in dwi_dict.iteritems() for i in range(0,len(dwi_dict['bval'])) if v[i] == dirnum }
+                    PEdirNums = set([dwi_dict['direction'][i] for i in idxs])
+                    for PEdirNum in PEdirNums:
+                        dwis = [ dwi_dict['dwiFile'][i] for i in idxs if dwi_dict['direction'][i] == PEdirNum ]
+                        assert len(dwis) <= 2
+                        dwiname = "Diffusion" + "_dir-" + str(dirnum) + "_" + session + "_corr_" + str(PEdirNum)
+                        if "j" in PEdirNum:
+                            PEdir = 2
+                        elif "i" in PEdirNum:
+                            PEdir = 1
+                        else:
+                            RuntimeError("Phase encoding direction not specified for diffusion data.")
+                        pos = "EMPTY"
+                        neg = "EMPTY"
+                        gdcoeffs = "None"
+                        for dwi in dwis:
+                            if "-" in layout.get_metadata(dwi)["PhaseEncodingDirection"]:
+                                neg = dwi
+                            else:
+                                pos = dwi
+                        try:
+                            echospacing = layout.get_metadata(pos)["EffectiveEchoSpacing"] * 1000
+                            dwi_stage_dict = OrderedDict([("DiffusionPreprocessing", partial(run_diffusion_processsing,
+                                                                                             posData=pos,
+                                                                                             negData=neg,
+                                                                                             path=args.output_dir,
+                                                                                             subject="sub-%s" % subject_label,
+                                                                                             echospacing=echospacing,
+                                                                                             PEdir=PEdir,
+                                                                                             gdcoeffs="NONE",
+                                                                                             dwiname=dwiname,
+                                                                                             n_cpus=args.n_cpus))])
+                            for stage, stage_func in dwi_stage_dict.iteritems():
+                                if stage in args.stages:
+                                    stage_func()
+                        except:
```

Comment:
Please replace this with a more specific catch (narrower in the amount of code it encomapses and types of exceptions it is suppose to catch). Right now it will catch all exceptions making debugging really hard.